mysticeti-core/src/commit_observer.rs

```diff
@@ -180,3 +180,47 @@ impl<H: ProcessedTransactionHandler<TransactionLocator> + Send + Sync> CommitObs
         self.commit_interpreter.committed = committed;
     }
 }
+
+#[derive(Default)]
+pub struct SimpleCommitObserver {
+    commit_interpreter: Linearizer,
+    committed_leaders: Vec<BlockReference>,
+}
+
+impl SimpleCommitObserver {
+    pub fn new() -> Self {
```

Comment:
Here we should also add parameters (which I think will be useful for @akichidis who are integrating this with sui) - the channel commit observer writes to, and the start replay index, which we will integrate later)